specify/specify7
# Batch Edit: Support for editing basic fields

Fixes #5413
Fixes #6209
Uses code from #4929

⚠️ **Note:** This PR affects database migrations. See [migration testing instructions](https://specify.github.io/testing/pull_requests.html#prs-tagged-with-label-migration).

This PR pretty much has the same code as #4929 but updated to production. To minimize the scope for testing, batch edit for relationships is disabled in this PR and will be enabled in the follow up PR - batch edit for relationships. There are still some merge conflicts resulting from some newer code in prod which will be resolved in the next PR.

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions

- Create a query
- Click on Batch Edit. This will create a new Batch Edit dataset with columns involving relationships disabled (readonly)
- Change values you wish to edit
- Click Validate
- [ ] Verify changed values are highlighted as updated cells
- Click Commit. This will apply your changes
- [ ] Re-run the original query and verify the values changed as expected
- Click rollback
- [ ] Verify values were rolled back to original

#### Regression tests
- [ ] Retest #5489
